src/ui.data-table/UTableRow.vue

```diff
@@ -18,19 +18,23 @@
       :class="cx([getCellAttrs(key, row, index).class, columns[index].tdClass])"
     >
       <div
-        v-if="(row.row || nestedLevel) && index === 0"
+        v-if="(row.row || nestedLevel || row.nestedData) && index === 0"
         :style="getNestedShift()"
         v-bind="bodyCellNestedAttrs"
       >
         <UIcon
-          v-if="row.row"
+          v-if="row.row || (row.nestedData && hasSlotContent($slots['nested-content']))"
           size="xs"
           internal
           interactive
-          :name="row?.row?.isHidden ? config.defaults.expandIcon : config.defaults.collapseIcon"
+          :name="
+            row?.row?.isHidden || row?.nestedData?.isHidden
```

Comment:
move into fucntion or computed if possible. first `?` in chains is no need i guess.